### DISCUSSION TITLE
How do i use pvc with the helm charts?

### DISCUSSION THREAD

**Discussion by encryptblockr** [strimzi/strimzi-kafka-operator#5838] [Q&A] [answered]
Looking at the helm charts for this kafka operator
with the persistent nature of kafka, kind of disappointed that there is no mention of persistent storage at all on docs
https://github.com/strimzi/strimzi-kafka-operator/tree/main/helm-charts/helm3/strimzi-kafka-operator

Where in the values.yaml file do i reference what persistent volume claim to use and what size???

**Comment by scholzj** [comment, marked as answer]
The Strimzi Helm Chart is only used to install the Strimzi Operator. Once it is installed, you tell it to configure a Kafka cluster by creating the Kafka custom resource which is where you configure the storage details. Check the examples (https://github.com/strimzi/strimzi-kafka-operator/tree/main/examples/kafka) and docs for more details.

**Reply by encryptblockr** [reply]
am not getting this

so if i want to deploy 2 kafka clusters for example, lets say `kafka-cluster-a` and `kafka-cluster-b`
do i install the operator for each cluster or just once in the k8s cluster and the then the  multiple kafka clusters use same operator?

**Reply by scholzj** [reply]
That really depends on you and on how you configure it. You can do both. You can have one operator watch your whole Kubernetes cluster or multiple namespaces and manage multiple different Kafka cluster in them. You can also deploy it to watch just a single namespace. The docs explain the details: https://strimzi.io/docs/operators/latest/deploying.html#cluster-operator-str

**Reply by encryptblockr** [reply]
what if there is an existing operator watching the whole cluster and i want to create another operator for a few specific namespaces, will the operators conflict?

**Reply by scholzj** [reply]
They will conflict and things will not work well. If you have operator watching the whole cluster, you should not install and other copies of it.

**Reply by encryptblockr** [reply]
ok cool..is there helm charts for installing kafka cluster itself? besides just bunch of yaml files?
if no, any reason why not?

**Reply by scholzj** [reply]
No - at least not an official one which would be part of Strimzi.

**Reply by scholzj** [reply]
Sorry, missed this part in the first reply ...

> if no, any reason why not?

I guess there are two reasons:
* In my personal view is that the `Kafka` custom resource is too complicated for what Helm can do. You would either have to always deal with options not being supported by the Helm Chart and would and up with huge amount of different options.
* The Helm Chart is mostly contributed by users and nobody contributed a Helm Chart for the custom resources

**Reply by encryptblockr** [reply]
hmm, not sure if there are anything special with the CRDs, as long as described in yaml, helm just deploys the yaml files it sees
or am i missing something?

**Reply by scholzj** [reply]
We are not talking about CRDs here => those are deployed by the operator Helm Chart. We talk about the Kafka custom resources which have quite a lot of different options and everyone uses different ones. So you either write the Helm Chart to customize just a few basic ones. But then it will not be usable for many users. Or you will end up having hundreds of different options to please everyone. So simply writing the YAML is much easier.

But that is just my personal view - I don't think it would be a problem if you or anyone else want to contribute it.

**Comment by brsolomon-deloitte** [comment]
@encryptblockr you could also check out the [Bitnami Helm charts for Kafka](https://github.com/bitnami/charts/tree/master/bitnami/kafka) for inspiration, but keep in mind those are not using the Strimzi CRDs - they are just regular built-in k8s object types.

**Reply by encryptblockr** [reply]
yeah aware of that helm charts...but wanted to know if strimzi has one specific to this project...after all they made helm charts for the operator, so wondering what went wrong in not making helm charts for the cluster deployment also
